### PR TITLE
Added GHA size check action

### PR DESCRIPTION
**Pull request by mkolesnik**
Also added linting check to Shipyard itself

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: gh-actions/commit-size/action.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - name: Make sure the size of each commit doesn't pass the desired size
+      shell: bash
+      run: |
+        set -e
+        git fetch origin refs/pull/${{ github.event.pull_request.number}}/head
+        commits_range=${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}
+        sum_commit() { git diff --numstat ${1} ${1}^ | cut -f1,2 -d$'\t' | tr $'\t' '+' | paste -sd+ - | bc; }
+        for commit in $(git log --pretty="format:%H" ${commits_range}); do
+            if [[ $(sum_commit ${commit}) -gt ${{ inputs.size }} ]]; then
+                 echo "Commit ${commit} is bigger than allowed size ${{ inputs.size }}."
```
Comment (tpantelis) [inline gh-actions/commit-size/action.yaml]: Can we also print the size of the offending commit?

```suggestion
                 size=sum_commit ${commit}
                 ...
                 echo "Size of commit ${commit} (${size}) is larger than allowed size ${{ inputs.size }}."
```

File: gh-actions/commit-size/action.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                 exit 1
+            fi
+
+            echo "Commit ${commit} is within allowed size ${{ inputs.size }}."
```
Comment (tpantelis) [inline gh-actions/commit-size/action.yaml]: Also print size here?

```suggestion
            echo "Size of commit ${commit} (${size})  is within allowed size ${{ inputs.size }}."
```

**Review by tpantelis — DISMISSED** [review]

File: gh-actions/commit-size/action.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Make sure the size of each commit doesn't pass the desired size
+      shell: bash
+      run: |
+        set -e
+        git fetch origin refs/pull/${{ github.event.pull_request.number}}/head
+        commits_range=${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }}
+        sum_commit() { git diff --numstat ${1} ${1}^ | cut -f1,2 -d$'\t' | tr $'\t' '+' | paste -sd+ - | bc; }
```
Comment (skitt) [inline gh-actions/commit-size/action.yaml]:
```suggestion
        sum_commit() { git diff --numstat ${1} ${1}^ | awk '{ sum += $1 + $2 } END { print sum }; }
```

**Review by skitt — APPROVED** [review]
Minor tweak, nothing blocking. It’s a shame `git` doesn’t have a “group by” ;-).

File: gh-actions/commit-size/action.yaml (hunk 3 above)
Comment (mkolesnik) [reply gh-actions/commit-size/action.yaml]: Thanks, you were missing a closing quote so I applied manually and force-pushed an updated version.

Comment (skitt) [reply gh-actions/commit-size/action.yaml]: Ah yes, apologies!

**Review by dfarrell07 — APPROVED** [review]

**Review by skitt — DISMISSED** [review]
It turns out this needs to deal with merge commits differently; see the failure after merging `devel`:

Screenshot from 2021-07-16 10-43-59

The commit that’s too large is the merge commit.

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**